File: INF-143/contest_5/A/main.cpp
#include<bits/stdc++.h>

using namespace std;

long long int izquierda(int n)
{
	return (n - 1) * (n) / 2;
}

long long int derecha(long long int n, long long int N)
{
	return (N * (N + 1) / 2) - (n * (n + 1) / 2);
}

bool f(long long int n, long long int N)
{
	return izquierda(n) == derecha(n, N);
}

int main()
{
	freopen("input.txt", "r", stdin);

	long long int n;
	
	while(cin >> n)
	{	
		long long int low = 1, high = n, mid;
		
		while(high - low > 1)
		{
			mid = (low + high) / 2;
			cout << "\t" << mid << "\n";
			cout << "\t\t" << izquierda(mid) << " " <<  derecha(mid, n) << "\n";
			if(izquierda(mid) <= derecha(mid, n))
				low = mid;
			else
				high = mid;

		}
		/*
		if(f(low, n))
			cout << low << "\n";
		else
			cout << "NO\n";*/
		cout << low << "\n";
		
	}

	return 0;
}

File: INF-143/contest_5/A/main.py
def izquierda(n):
	return (n - 1) * (n) / 2;

def derecha(n, N):
	return (N * (N + 1) / 2) - (n * (n + 1) / 2)

def f(n, N):
	return izquierda(n) == derecha(n, N)

def sol():
	n = int(input())
	low = 1; high = n;
		
	while high - low > 1 :

		mid = (low + high) // 2;
		
		if izquierda(mid) <= derecha(mid, n):
			low = mid
		else:
			high = mid
	if f(low, n):
		print(low)
	else:
		print("NO")

while True:
	try:
		sol()
	except EOFError:
		break